### ISSUE TITLE
get_type() don't return real array-size

### ISSUE THREAD

**Issue by twanks123** [KhronosGroup/SPIRV-Cross#30]
Hello guys, the following function doesn't return the real array size.
Here is my code:
[SPIRV]
`layout (set=0, binding=1) uniform texture2D myArrayOfSampledImages[12];`

[Code]
`       
        
        for (auto &resource : resources.storage_images)
        {
            unsigned int setNum = comp.get_decoration(resource.id, spv::DecorationDescriptorSet);
            unsigned int bindingNum = comp.get_decoration(resource.id, spv::DecorationBinding);

            spirv_cross::SPIRType type = comp.get_type(resource.type_id);
        }
`

![spirv_cross](https://cloud.githubusercontent.com/assets/12892928/16779404/5fbf1d6c-4872-11e6-9d6e-7806d92be72a.png)

Or is this the wrong way to get the total array size?^^

**Comment by HansKristian-ARM** [comment]
Hi. I found that the type_id that was returned wasn't suited for parsing this information, so I've pushed up a commit which adds both the type_id and base_type_id to the reflection interface.

Unfortunately SPIR-V will have the interesting meta-data in the base_type and all array information on top of that will be a different type_id.

Your code there should work now (although I'd suggest capturing the type by const reference instead of copy).
